(DOCSP-27483) Adds guidance to create Atlas search index from a file (#146)
mongodb/docs-atlas-cli: source/reference/json/search-index-config-file.txt

.. _atlas-cli-search-index-config-file:

=====================================
Atlas Search Index Configuration File
=====================================

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

> You can use an |fts| configuration file to specify the settings
> required when you :ref:`create a search index
> <atlas-clusters-search-indexes-create>` using the {+atlas-cli+}. The 

Reviewer: ```suggestion
You can use an |fts| index configuration file to specify the required 
settings for :ref:`creating a search index
<atlas-clusters-search-indexes-create>` using the {+atlas-cli+}. The 
```

{+atlas-cli+} accepts ``.json`` search index configuration files.

.. _search-index-settings:

Atlas Search Index Settings
---------------------------

You can specify the following settings in the |fts| index
configuration file. For a full list of available settings, see
the request body schema in the API specification:
:oas-atlas-op:`Create One Atlas Search Index </createAtlasSearchIndex>`.

.. list-table:: 
   :header-rows: 1 
   :widths: 20 10 70 

   * - Field 
     - Type 
     - Description 

   * - ``collectionName``
     - string

>      - Label that identifies the collection that contains one or more
>        |fts| indexes.

Reviewer: ```suggestion
     - Label that identifies the collection for which you want to create an 
       |fts| index.
```

   * - ``database``
     - string

>      - Label that identifies the database that contains the collection
>        with one or more |fts| indexes.

Reviewer: ```suggestion
     - Label that identifies the database that contains the collection
       for which you want to create an |fts| index.
```

   * - ``name``
     - string
     - Label that identifies this index. Within each namespace, names
       of all indexes in the namespace must be unique.

   * - ``mappings``
     - object
     - Index specifications for the collection's fields.

   * - ``mappings.dynamic``
     - boolean
     - Flag that indicates whether the index uses dynamic or static

>        mappings. Required if ``mappings.fields`` is omitted. To learn
>        more, see :atlas:`atlas-search/index-definitions`.

Reviewer: Dynamic is optional as it defaults to true if omitted. Suggest:
```suggestion
       mappings. If omitted or if set to ``false``, ``mappings.fields`` is required. To learn
       more, see :atlas:`Atlas Search Index Syntax </atlas-search/index-definitions/>`.
```
